### PR TITLE
Feature 0.4.0 range

### PR DESCRIPTION
**Pull request by mkatychev**
* range is added to recordings: `dark record --range "<start_num>:<end_num>" ./dir/ my_reel_name`, allowing one to limit  what frames run in a record session
* `grpcurl` errors propagate to stdout properly

### CODE REVIEW

File: filmreel/src/reel.rs [hunk 1]
```diff
@@ -57,6 +56,32 @@ impl Reel {
                 .collect(),
         }
     }
+
+    /// get_metaframes takes a directory glob ref and a possible range, returning a vector of
+    /// MetaFrames
+    fn get_metaframes<T>(dir_glob: T, range: Option<Range<u32>>) -> Result<Vec<MetaFrame>, FrError>
```
Comment (mkatychev) [inline filmreel/src/reel.rs:62]: I wonder how exactly I can test this function, perhaps split it out into two smaller functions? 🤔

File: src/http.rs [hunk 2]
```diff
@@ -49,19 +49,42 @@ pub fn build_request(prm: Params, req: Request) -> Result<RequestBuilder, Error>
         Err(e) => return Err(Error::from(e)),
     }
 
-    let form = serde_json::from_value(req.get_etc()["form"].clone())?;
-    match form {
-        Value::Object(_) => builder = builder.query(&form),
-        Value::Null => {}
+    // TODO dedupe these two parts
```
Comment (mplanchard) [inline src/http.rs:52]: 🤷 this doesn't seem to me sufficient duplication to warrant an abstraction, unless it's going to give us better unit-testing or something

File: src/record.rs [hunk 3]
```diff
@@ -148,3 +153,53 @@ fn parse_component(component: String) -> Result<(Reel, Register), Error> {
...
+type ParsedRange = Option<Range<u32>>;
+// parse_range parses the `"<start_u32>:<end_u32>"` provided to the `--range` cli argument
+// returning a range object
+fn parse_range<T>(str_range: T) -> Result<ParsedRange, Error>
+where
+    T: AsRef<str>,
+{
+    match str_range
+        .as_ref()
+        .splitn(3, ':')
```
Comment (mplanchard) [inline src/record.rs:166]: curious why 3 here if we're expecting at most 2 range args? Could this lead to potentially silent unexpected behavior if someone provided something like `12:34:-1` expecting the last argument to be a `step` operator, due to potentially ignoring the argument due to the `_ => Ok(None)` below?

Comment (mkatychev) [reply src/record.rs:166]: Good catch, that was a mistake!

**Pull request merged**